weblate/addons/models.py

```diff
@@ -208,34 +209,37 @@ def handle_addon_error(addon, component):
         addon.disable()
 
 
-@receiver(vcs_pre_push)
-def pre_push(sender, component, **kwargs):
-    for addon in Addon.objects.filter_event(component, EVENT_PRE_PUSH):
-        component.log_debug("running pre_push add-on: %s", addon.name)
+def handle_addon_event(sender, component, event_type, **kwargs):
+    """Executing addon event for a particular event type."""
+    event_string = EVENT_STRING.get(event_type)
+    translation = kwargs.get("translation")
+    for addon in Addon.objects.filter_event(component, event_type):
+        translation.log_debug(f"running {event_string} add-on: {addon.name}")
         try:
-            with sentry_sdk.start_span(op="addon.pre_push", description=addon.name):
-                addon.addon.pre_push(component)
+            with sentry_sdk.start_span(
+                op=f"addon.{event_string}", description=addon.name
+            ):
+                addon_object = addon.addon
+                if hasattr(addon_object, event_string):
+                    method = getattr(addon_object, event_string)
+                    method(**kwargs)
+
         except DjangoDatabaseError:
             raise
         except Exception:
             handle_addon_error(addon, component)
         else:
-            component.log_debug("completed pre_push add-on: %s", addon.name)
+            translation.log_debug(f"completed {event_string} add-on: {addon.name}")
+
+
+@receiver(vcs_pre_push)
+def pre_push(sender, component, **kwargs):
+    handle_addon_event(sender, component, EVENT_PRE_PUSH, translation=component)
 
 
 @receiver(vcs_post_push)
 def post_push(sender, component, **kwargs):
-    for addon in Addon.objects.filter_event(component, EVENT_POST_PUSH):
-        component.log_debug("running post_push add-on: %s", addon.name)
-        try:
-            with sentry_sdk.start_span(op="addon.post_push", description=addon.name):
-                addon.addon.post_push(component)
-        except DjangoDatabaseError:
-            raise
-        except Exception:
-            handle_addon_error(addon, component)
-        else:
-            component.log_debug("completed post_push add-on: %s", addon.name)
+    handle_addon_event(sender, component, EVENT_POST_PUSH, translation=component)
```

Comment:
`translation=component` looks weird. What is the intention here?

Reply:
handle_addon_event body says -

"""
translation = kwargs.get("translation")
 for addon in Addon.objects.filter_event(component, event_type):
        translation.log_debug(f"running {event_string} add-on: {addon.name}")
"""

and post_push function is earlier handled as 
"""    
for addon in Addon.objects.filter_event(component, EVENT_POST_PUSH):
    component.log_debug("running post_push add-on: %s", addon.name)
"""
hance I am sending component in translation.